server/src/services/ito/itoService.ts

```diff
@@ -200,68 +259,67 @@ export default (router: ConnectRouter) => {
           ? HeaderValidator.validateVocabulary(vocabularyHeader)
           : []
 
-        const asrModelHeader = context.requestHeader.get('asr-model')
-        // Use header value if provided, otherwise fall back to environment variable for backwards compatibility
-        const asrModelToValidate =
-          asrModelHeader || process.env.GROQ_TRANSCRIPTION_MODEL
-
-        if (!asrModelToValidate) {
-          throw new ConnectError(
-            'ASR model must be provided either in header or GROQ_TRANSCRIPTION_MODEL environment variable',
-            Code.InvalidArgument,
-          )
-        }
-
-        const asrModel = HeaderValidator.validateAsrModel(asrModelToValidate)
-        console.log(
-          `[Transcription] Using validated ASR model: ${asrModel} (source: ${asrModelHeader ? 'header' : 'env'})`,
+        const advancedSettingsHeaders = getAdvancedSettingsHeaders(
+          context.requestHeader,
         )
 
         // 4. Send the corrected WAV file using the validated ASR model from headers.
-        let transcript = await groqClient.transcribeAudio(
-          fullAudioWAV,
-          'wav',
-          asrModel,
-          vocabulary,
-        )
-
-        console.log(
-          `📝 [${new Date().toISOString()}] Received transcript: "${transcript}"`,
-        )
-
-        // 5. Check if transcript contains "Hey Ito" in the first 5 words
-        const words = transcript.trim().split(/\s+/)
-        const firstFiveWords = words.slice(0, 5).join(' ').toLowerCase()
-
-        let mode = ItoMode.TRANSCRIBE
-        if (firstFiveWords.includes('hey ito')) {
-          mode = ItoMode.EDIT
+        let transcript
+        switch (advancedSettingsHeaders.asrProvider) {
+          case 'groq':
+            console.log('Transcribing with Groq ASR provider')
+            transcript = await transcribeWithGroq(
+              fullAudioWAV,
+              advancedSettingsHeaders,
+              vocabulary,
+            )
+            break
+          default:
+            console.log('No provider matched, using Groq by default')
+            transcript = await transcribeWithGroq(
+              fullAudioWAV,
+              advancedSettingsHeaders,
+              vocabulary,
+            )
+            break
```

Comment:
I'd rather push this switch statement lower, closer to where the actual call to transcription is happening. I think you can drop these changes as part of your pr as I'll probably have to overwrite them anyway with cerebras support

Reply:
Same with the adjustTranscript switch block -- you can just drop those changes if you want